dataplicity/device_meta.py

```diff
@@ -43,15 +48,29 @@ def get_os_version():
     distro = " ".join(platform.linux_distribution()).strip()
     return distro
 
+
 def get_ip_address_list():
     # Get the ip addresses from all the interfaces
     try:
         interfaces = get_all_interfaces()
     except Exception:
-        log.exception('unable to retrieve interface information')
+        log.exception("unable to retrieve interface information")
         # Sorry for the pokemon exception, but I don't know how
         # reliable the call is, and if it fails what it will fail with.
         # It needs some exception handling or the whole get_meta call
         # will fail
         return []
     return [i[1] for i in interfaces]
+
+
+def get_tag_list():
+    """Run the taglist.sh script, get output as a list of tags"""
+    try:
+        output = subprocess.check_output(TAG_SCRIPT)
+    except (OSError, subprocess.CalledProcessError) as error:
+        log.error(error)
+        return []
+
+    # regex split on comma, spaces, newline and tabs
+    tag_list = re.split(r"[,\s\n\t]", output)
+    return [tag for tag in tag_list if tag != ""]
```

Comment:
Maybe strip the tags in case there is extra whitespace.

Bear in mind that this is Py2.7 and these strings are byte strings. They will need to be converted to unicode at some point, with all the pitfalls that involves.

Also bear in mind that this is untrusted data. And you probably want to sanitize it somewhere. Maybe not in the agent, but on the server.

Reply:
The server side validates the tag strings against a regex, so that's handled.

We can cover in the user docs what is expected in these strings.  They can ignore that of course, but it means we can enforce ascii characters for example, making conversion easier.